src/ops/ops.go

```diff
@@ -124,6 +125,71 @@ func (o *ops) SystemctlAction(action string, args ...string) error {
 	return errors.Wrapf(err, "Failed executing systemctl %s %s", action, args)
 }
 
+var ostreeOutputRegex = regexp.MustCompile(`Imported: (\w+)`)
+
+func (o *ops) importOSTreeCommit(liveLogger io.Writer) (string, error) {
+	ostreeReleasePullSpec := fmt.Sprintf("ostree-unverified-registry:%s", o.installerConfig.CoreosImage)
+	out, err := o.ExecPrivilegeCommand(liveLogger, "ostree", "container", "unencapsulate", "--authfile", "/root/.docker/config.json", "--quiet", "--repo", "/ostree/repo", ostreeReleasePullSpec)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to unencapsulate rhcos payload image: %s", out)
+	}
+
+	matches := ostreeOutputRegex.FindStringSubmatch(out)
+	if matches == nil {
+		return "", fmt.Errorf("got unexpected output from unencapsulate: \"%s\"", out)
+	}
+
+	return matches[1], nil
+}
+
+func (o *ops) WriteImageToLocalDevice(liveLogger io.Writer, ignitionPath string) error {
+	out, err := o.ExecPrivilegeCommand(liveLogger, "mount", "/sysroot", "-o", "remount,rw")
+	if err != nil {
+		return errors.Wrapf(err, "failed to remount sysroot: %s", out)
+	}
+	out, err = o.ExecPrivilegeCommand(liveLogger, "mount", "/boot", "-o", "remount,rw")
+	if err != nil {
+		return errors.Wrapf(err, "failed to remount boot: %s", out)
+	}
+
+	out, err = o.ExecPrivilegeCommand(liveLogger, "ostree", "admin", "stateroot-init", "install")
+	if err != nil {
+		return errors.Wrapf(err, "failed creating new stateroot: %s", out)
+	}
+
+	commit, err := o.importOSTreeCommit(liveLogger)
+	if err != nil {
+		return err
+	}
+	o.log.Infof("imported commit %s", commit)
+
+	out, err = o.ExecPrivilegeCommand(liveLogger, "ostree", "admin", "deploy",
+		"--stateroot", "install",
+		"--karg", "$ignition_firstboot",
+		"--karg", defaultIgnitionPlatformId,
+		commit)
+	if err != nil {
+		return errors.Wrapf(err, "failed to deploy commit to stateroot: %s", out)
+	}
```

Comment:
Related to https://github.com/openshift/assisted-installer/pull/1003#discussion_r1913421655 ...

@eranco74 (or someone you think knows) can/should we use `bootc` here?

Reply:
Bootc is already part of rhcos?

Reply:
I'm not sure, that's part of what I'm asking, but I can check.

Reply:
Yeah, seems to be there in 4.18 at least 
![image](https://github.com/user-attachments/assets/0ba43c41-bc85-4835-8a93-c6d93b112cf6)

Reply:
It's also present in 4.17 and 4.16. 4.15 is the latest version that doesn't have it.

Reply:
I think MCE won't support anything earlier than 4.16 by the time this change gets in so we might be safe there, but if we also want to use this on the SaaS it might not be worth limiting the versions we can install.

Even if `bootc` works for this and is the better option maybe we can leave it as a follow up for now?

Reply:
Leaving this as a potential follow up